packages/chain-adapters/src/ChainAdapterManager.ts

```diff
@@ -81,15 +99,19 @@ export class ChainAdapterManager {
    * @param {ChainTypes} chain - Coin/network symbol from Asset query
    * @param {Function} factory - A function that returns a ChainAdapter instance
    */
-  addChain<T extends ChainTypes>(chain: T, factory: () => ChainAdapter<ChainTypes>): void {
+  addChain<T extends SupportedChainId>(
+    chain: T,
+    factory: () => ChainAdapter<SupportedChainId>
+  ): void {
     if (typeof chain !== 'string' || typeof factory !== 'function') {
       throw new Error('Parameter validation error')
     }
     this.supported.set(chain, factory)
   }
 
-  removeChain<T extends ChainTypes>(chain: T): void {
-    if (!Object.values(ChainTypes).includes(chain)) {
+  removeChain<T extends SupportedChainId>(chain: T): void {
+    // FIXME: This is gross
```

Comment:
do we really even need this? T is already constrained to `SupportedChainId` and seem redundant (and gross as mentioned) to do this additional check

Reply:
Yeah this logic should live in a type guard, but if you move it to a type guard, you realize it should always return true according to the typings

Reply:
We killed `ChainAdapterManager` entirely - problem solved!